akka-stream/src/main/scala/akka/stream/StreamTimeoutException.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import scala.concurrent.TimeoutException
+import scala.util.control.NoStackTrace
+import akka.annotation.DoNotInherit
+
+/**
+ * Base class for timeout exceptions specific to Akka Streams
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed class StreamTimeoutException(msg: String) extends TimeoutException(msg) with NoStackTrace
+
+final class InitialTimeoutException(msg: String) extends StreamTimeoutException(msg)
+
+final class CompletionTimeoutException(msg: String) extends StreamTimeoutException(msg)
+
+final class IdleTimeoutException(msg: String) extends StreamTimeoutException(msg)
```

Comment:
There is already an `TcpIdleTimeoutException` , should this be `StreamIdleTimeoutException`?

Reply:
Makes sense to be explicit 👍

Reply:
How about the others?

Reply:
@patriknw Should we revert it back to `IdleTimeoutException`?

Reply:
I see, let's keep it as is then. `StreamIdleTimeoutException`

Reply:
Okay, let me update the PR to the final state.